full-service/src/service/balance.rs

```diff
@@ -266,21 +262,13 @@ where
     fn get_balance_inner(
         account_id_hex: Option<&str>,
         assigned_subaddress_b58: Option<&str>,
+        token_id: TokenId,
         conn: &Conn,
-    ) -> Result<(u128, u128, u128, u128, u128, u128, u128), BalanceServiceError> {
-        let max_spendable = Txo::list_spendable(
-            account_id_hex,
-            None,
-            assigned_subaddress_b58,
-            *Mob::ID,
-            conn,
-        )?
-        .max_spendable_in_wallet;
-
+    ) -> Result<Balance, BalanceServiceError> {
         let unspent = sum_query_result(Txo::list_unspent(
             account_id_hex,
             assigned_subaddress_b58,
-            Some(0),
+            Some(*token_id),
```

Comment:
what is the `*` doing here? according to SO it de-references the value? Not sure what that means or why it's being used here